jsonschema-core.xml

```diff
@@ -581,39 +554,61 @@
                     This value SHOULD be normalized, and SHOULD NOT be an empty fragment &lt;#&gt;
                     or an empty string &lt;&gt;.
                 </t>
-                <t>
-                    The root schema of a JSON Schema document SHOULD contain an "$id" keyword with
-                    a URI (containing a scheme).  This URI SHOULD either not have a fragment, or
-                    have one that is an empty string.
-                    <!-- All of the standard meta-schemas use an empty fragment in their id/$id values. -->
-                    <cref>
-                        How should an "$id" URI reference containing a fragment with other components
-                        be interpreted?  There are two cases:  when the other components match
-                        the current base URI and when they change the base URI.
-                    </cref>
-                </t>
-                <t>
-                    To name subschemas in a JSON Schema document,
-                    subschemas can use "$id" to give themselves a document-local identifier.
-                    This is done by setting "$id" to a URI reference consisting
-                    only of a plain name fragment (not a JSON Pointer fragment).
-                    The fragment identifier MUST begin with a letter ([A-Za-z]), followed by
-                    any number of letters, digits ([0-9]), hyphens ("-"), underscores ("_"), colons
-                    (":"), or periods (".").
-                </t>
-                <t>
-                    Providing a plain name fragment enables a subschema to be
-                    relocated within a schema without requiring that JSON
-                    Pointer references are updated.
-                </t>
-                <t>
-                    The effect of defining a fragment-only "$id" URI reference that neither
-                    matches the above requirements nor is a valid JSON pointer
-                    is not defined.
-                </t>
-                <t>
-                    For example:
+                <section title="Identifying the root schema">
+                    <t>
+                        The root schema of a JSON Schema document SHOULD contain an "$id" keyword with
+                        a URI (containing a scheme).  This URI SHOULD either not have a fragment, or
+                        have one that is an empty string.
```

Comment:
This is probably better suited for a follow-up PR, but this is sort of confusing to parse, maybe

> This URI SHOULD have no fragment, or an empty fragment.

Maybe we simplify the entire paragraph down to:

> The root schema of a JSON Schema document SHOULD contain an "$id" keyword with a base URI [RFC3986] (a full URI with a scheme but no fragment), or a full URI with an empty fragment.